addons/analytic/models/analytic_mixin.py

```diff
@@ -36,40 +32,50 @@ def init(self):
             self.env.cr.execute(query)
         super().init()
 
-    @api.model
-    def fields_get(self, allfields=None, attributes=None):
-        """ Hide analytic_distribution_search from filterable/searchable fields"""
-        res = super().fields_get(allfields, attributes)
-        if res.get('analytic_distribution_search'):
-            res['analytic_distribution_search']['searchable'] = False
-        return res
-
     def _compute_analytic_distribution(self):
         pass
 
-    def _search_analytic_distribution(self, operator, value):
-        if operator not in ['=', '!=', 'ilike', 'not ilike'] or not isinstance(value, (str, bool)):
+    def _leaf_to_sql(self, alias: str, leaf: tuple, query: Query) -> SQL:
+        left, operator, value = leaf
+        # Don't use this override when account_report_analytic_groupby is truely in the context
+        # Indeed, when account_report_analytic_groupby is there it means that `analytic_distribution`
+        # doesn't have the same format and the table is a temporary one, see _prepare_lines_for_analytic_groupby
+        if left != 'analytic_distribution' or self.env.context.get('account_report_analytic_groupby'):
+            return super()._leaf_to_sql(alias, leaf, query)
+
+        if operator not in ('=', '!=', 'ilike', 'not ilike', 'in', 'not in'):
             raise UserError(_('Operation not supported'))
-        operator_name_search = '=' if operator in ('=', '!=') else 'ilike'
-        account_ids = list(self.env['account.analytic.account']._name_search(name=value, operator=operator_name_search))
 
-        query = f"""
-            SELECT id
-            FROM {self._table}
-            WHERE analytic_distribution ?| array[%s]
-        """
-        operator_inselect = 'inselect' if operator in ('=', 'ilike') else 'not inselect'
-        return [('id', operator_inselect, (query, [[str(account_id) for account_id in account_ids]]))]
+        if isinstance(value, str) and operator in ('=', '!=', 'ilike', 'not ilike'):
```

Comment:
You removed the case for a search if Analytic Distribution is set, so with a bool.
Btw, just re-adding it here in the condition won't be enough (it brings the opposite result, is set will bring the ones with an empty distribution).

Reply:
Thanks for the review, I fixed all your comments.